src/main/java/net/minecraftforge/snowblower/Generator.java

```diff
@@ -241,30 +309,39 @@ else if (snapshot == null)
                 attemptPush("Pushing 10 versions to remote.");
             }
         }
-        attemptPush("Pushing remaining versions to remote."); // In case there are any remaining versions to push
 
-        if (toGenerate.isEmpty())
-            this.logger.accept("No versions to process");
+        // TODO: Push commits in increments of 10 even if we generated more than 10 versions on a previous run of SnowBlower but didn't push then
```

Comment:
In order to batch previous commits you can find the last common commit between the local branch and remote branch, and work the way up the commit list 10 by 10 commits and pushing the 10th. Pushing a commit makes sure all commits before it are pushed to.